drivers/spi/spi_ll_stm32_fifo.c

```diff
@@ -0,0 +1,531 @@
+/*
```

Comment:
Why did this file get created?

Reply:
This driver supports 2 compatibles and so this file was created for 2 demo purposes:
-Demonstrate one example of supporting 2 different compatibles for a same IP type in a family
-Demonstrate the use of 2 different generation API (spi_ll_stm32_fifo.c  is using declare.py and spi_ll_stm32.c is using declare_template.py.

declare.py is providing a python style api:
```
* device_config = ( ['spi_stm32_config',
*       {'field': '.spi', 'value': '@reg/0', 'cast': '(SPI_TypeDef *)'},
*       {'field': '.pclken.bus', 'value': '@clock/bus'},
*       {'field': '.pclken.enr', 'value': '@clock/bits'},
*       {'string': ['.irq_config = st_stm32_spi_fifo_', 'index', '_config_irq'], 'flag': 'CONFIG_SPI_STM32_INTERRUPT'}]
* )
```
while declare_template.py was made later to propose a more friendly template style api:
```
* device_config = ( 'spi_stm32_config',
* """
*      .spi = (SPI_TypeDef *)${@reg/0},
*      .pclken.bus = ${@clock/bus},
*      .pclken.enr = ${@clock/bits},
* #ifdef CONFIG_SPI_STM32_INTERRUPT
*      .irq_config = st_stm32_spi_${node_index}_config_irq,
* #endif
* """)
```

The later seems more adapted and should be preferred, to my opinion.
In the end, there is no need for duplicating the files as device_declare supports multiple compatibles for a single driver (as demonstyrated for I2C STM32 and LSM6DSL)